scribble-lib/scribble/private/manual-style.rkt

```diff
@@ -218,27 +219,25 @@
       (let loop ([i i])
         (cond
           [(string? i)
-           (cond
-             [(regexp-match #px"^(.*)_([a-zA-Z0-9]+)(.*)$" i)
-              => (lambda (m)
-                   (append (loop (cadr m))
-                           (list (make-element 'subscript
-                                               (loop (caddr m))))
-                           (loop (cadddr m))))]
-             [(regexp-match #px"^(.*)\\^([a-zA-Z0-9]+)(.*)$" i)
-              => (lambda (m)
-                   (append (loop (cadr m))
-                           (list (make-element 'superscript
-                                               (loop (caddr m))))
-                           (loop (cadddr m))))]
-             [(regexp-match #px"^(.*)([()0-9{}\\[\\]\u03C0])(.*)$" i)
-              => (lambda (m)
-                   (append (loop (cadr m))
-                           (list (caddr m))
-                           (loop (cadddr m))))]
-             [else
-              (list (make-element 'italic (list i)))])]
-          [(eq? i 'rsquo) (list 'prime)]
+           (match i
+             [(or (pregexp #px"^(.*)(_|\\^)\\{(.*)\\}(.*)$"
```

Comment:
I'm actually skeptical now that you can do this properly with regexp, since braces are supposed to be balanced. 

Take `A_{B^{C}}` as an example. With the current regexp, you would get `"A_{B"` as the first part and `"C}"` as the second part.